### PR TITLE
Introduce QT5_GENTOO_PRIVATE_CONFIG

### PR DESCRIPTION
**Pull request by kensington**
This behaves similarly to QT5_GENTOO_CONFIG, but for the new qmodule.pri instead.

### CODE REVIEW

File: eclass/qt5-build.eclass [hunk 1]
```diff
@@ -859,4 +898,61 @@ qt5_regenerate_global_qconfigs() {
 	else
 		ewarn "${qconfig_pri} or ${qconfig_pri_orig} does not exist or is not a regular file"
 	fi
+
+	if [[ ${QT5_MINOR_VERSION} -ge 9 ]]; then
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: minor thing: split this whole new piece of code into a separate function? then you can call it only in 5.9 and later, which reduces the levels of indentation inside the new function.

File: eclass/qt5-build.eclass [hunk 2]
```diff
@@ -859,4 +898,61 @@ qt5_regenerate_global_qconfigs() {
...
+		einfo "Updating QT.global_private in qmodule.pri"
+
+		local qmodule_pri=${ROOT%/}${QT5_ARCHDATADIR}/mkspecs/qmodule.pri
+		local qmodule_pri_orig=${ROOT%/}${QT5_ARCHDATADIR}/mkspecs/gentoo/qmodule-qtcore.pri
+		if [[ -f ${qconfig_pri} && -f ${qconfig_pri_orig} ]]; then
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: s/qconfig/qmodule/g

File: eclass/qt5-build.eclass [hunk 3]
```diff
@@ -765,12 +776,14 @@ qt5_qmake() {
 # @FUNCTION: qt5_install_module_qconfigs
 # @INTERNAL
 # @DESCRIPTION:
-# Creates and installs gentoo-specific ${PN}-qconfig.{h,pri} files.
+# Creates and installs gentoo-specific ${PN}-qconfig.{h,pri} and
+# ${PN}-qmodule.pri files.
 qt5_install_module_qconfigs() {
```
[the remark this answers is not in the document]
Comment (Pesa) [reply eclass/qt5-build.eclass]: or use a separate function like I suggested below

[the remark this answers is not in the document]
Comment (Pesa) [reply eclass/qt5-build.eclass]: OTOH, the version conditional is only temporary until we remove < 5.9, so whatever I'm fine either way

**Comment by kensington** [comment]
Thanks, I've landed this in 82026b039b4bc8ddc597c8186b464dd08f6339e3...e553539e7ee050d094da9a4673b2e5ac71aae422 with your feedback included.

I'll let this bake for a little bit, but in the absence of any issues or further feedback, I'd like to bump 5.9.4 soon with this.

**Comment by Pesa** [comment]
Agreed.

**Pull request closed without merging**